### DISCUSSION TITLE
Special characters are not correctly encoded if utf8 is only set in the config file

### DISCUSSION THREAD

**Discussion by swtrse** [openssl/openssl#22749] [Q&A] [answered]
Version: OpenSSL 3.0.12 24 Oct 2023 (Library: OpenSSL 3.0.10 1 Aug 2023)

Steps to follow:
myconfig.conf
```
[req]
distinguished_name  = req_distinguished_name
x509_extensions     = v3_req
prompt              = no
string_mask         = utf8only

[req_distinguished_name]
C                   = AT
ST                  = NÖ
L                   = Großmünster
O                   = Lübecker GmbH
OU                  = Firewall
CN                  = openwrt.lan

[v3_req]
keyUsage            = nonRepudiation, digitalSignature, keyEncipherment
extendedKeyUsage    = serverAuth
subjectAltName      = @alt_names

[alt_names]
DNS.1               = openwrt.lan
IP.1                = 10.0.0.1
IP.2                = fd18:222d:d95::1
```
If you create the cert with command
```
openssl req -x509 -nodes -days 3650 -newkey ec:ec.param -keyout mycert.key -out mycert.crt -config myconfig.conf
```
Special chars contained in the certificate are not encoded correctly.

The correct encoding can be achieved by
```
openssl req -x509 -nodes -utf8 -days 3650 -newkey ec:ec.param -keyout mycert.key -out mycert.crt -config myconfig.conf
```
The expected behavior is that `string_mask         = utf8only` is already taking care of the encoding which is not the case.

**Comment by t8m** [comment, marked as answer]
You need to also set `utf8 = yes` option. The `string_mask` option is related but different. See the req(1) manual page.

**Reply by swtrse** [reply]
Thanks. I did not see that option even when I looked. The solution is in fact working. The tutorial I was following seems to miss that one too.
